forest/drivers/argo.py

```diff
@@ -0,0 +1,163 @@
+        if pattern is None:
+            self.paths = []
+        else:
+            self.paths = glob.glob(pattern)
+
+    def navigator(self):
+        return Navigator()
+
+    def map_view(self, color_mapper):
+        return MapView(self.paths[0], color_mapper)
+
+    def profile_view(self, figure):
+        return ProfileView(self.paths[0], figure)
+
+
+class Navigator:
+    def initial_times(self, *args, **kwargs):
+        return []
+
+    def valid_times(self, *args, **kwargs):
+        return []
+
+    def variables(self, *args, **kwargs):
+        return ["temperature",]
+
+    def pressures(self, *args, **kwargs):
+        return []
+
+
+class MapView(Observable):
+    def __init__(self, path, color_mapper):
+        self.path = path
+        self.color_mapper = color_mapper
+        with netCDF4.Dataset(self.path) as dataset:
+            self.lons = dataset.variables["LONGITUDE"][:]
+            self.lats = dataset.variables["LATITUDE"][:]
+            self.surf_temp = dataset.variables["TEMP"][:, 0].data
+        self.source = bokeh.models.ColumnDataSource({
+            "x": [],
+            "y": [],
+            "surf_temp": [],
+        })
+        # update colour mapper with correct range?
+        self.color_mapper.low = np.amin(self.surf_temp)
+        self.color_mapper.high = np.amax(self.surf_temp)
+
+        # selected indices changes on the data source on a tap
+        self.source.selected.on_change("indices", self.update_profile_ids)
+        super().__init__()
+
+    def update_profile_ids(self, a, o, n):
+        with netCDF4.Dataset(self.path) as dataset:
+            plat_ids = []
+            for i in n:
+                plat_id = dataset.variables["PLATFORM_NUMBER"][i,:]
+                plat_ids.append(str(netCDF4.chartostring(plat_id)))
+
+        self.notify(set_profile_ids(n, plat_ids))
+
+    def add_figure(self, figure):
+        # Tap event listener
+        renderer = figure.circle(x="x", y="y", size=10,
+                                 color={'field': 'surf_temp',
+                                        'transform': self.color_mapper},
+                                 source=self.source)
+        figure.add_tools(bokeh.models.TapTool(renderers=[renderer]))
+        return renderer
+
+    def render(self, *args, **kwargs):
+        x, y = forest.geo.web_mercator(self.lons, self.lats)
+        self.source.data = {
+            "x": x,
+            "y": y,
+            "surf_temp": self.surf_temp,
+        }
+
+    def connect(self, store):
+        self.add_subscriber(store.dispatch)
+
+
+class ProfileView:
+    def __init__(self, path, figure):
+        self.path = path
+        self.figure = figure
+        self.source = bokeh.models.ColumnDataSource({
+            "x": [],
+            "y": [],
+        })
+        self.figure.circle(x="x", y="y", source=self.source)
+
+    def connect(self, store):
+        store.add_subscriber(self.render)
+
+    def render(self, state):
```

Comment:
The thinking for `Gallery.render` is to call `ProfileView.render_id(state, layer_id)` so that the individual profile view knows which settings it should match `forest.state.State.layers.index[layer_id]` returns a `dict` representation of layer specific settings, e.g. `variable` and `colorbar` settings for now